### ISSUE TITLE
Height parameter enforces new line switch

### ISSUE THREAD

**Issue by seqizz** [junegunn/fzf#2446]
- [X] I have read through the manual page (`man fzf`)
- [X] I have the latest version of fzf
- [X] I have searched through the existing issues

## Info

- OS
    - [X] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [X] zsh
    - [ ] fish
    
## Problem / Steps to reproduce

Hi there,

First of all, thank you for this tool. Amazingly helpful. Just wanted to report a small inconsistency.

Consider this hacky function on my .zshrc:
```
function _fuzzy_search_current_word() {
    CURRENTWORD="${LBUFFER/* /}${RBUFFER/ */}"
    # If there is no word entered, do nothing
    [[ -z $CURRENTWORD ]] && return
    # Pass current subtree to fzf
    FZFRESULT=`rg --files --no-ignore-vcs --hidden | fzf --reverse -q $CURRENTWORD`
    # Replace the word with selected result
    BUFFER=`echo "$BUFFER" | sed "s#${CURRENTWORD}#${FZFRESULT}#"`
    zle end-of-line
}
bindkey '^[^M' _fuzzy_search_current_word
```

Basically what it does is, when I type `vim somefuzzystring` and trigger this function with Alt-Enter, fzf opens and once I select the result, it replaces "somefuzzystring" with my selection (on the current line).

But when `--height` parameter is given to fzf, it prints the word on the next line for some reason. None of the options I've tried changed this behavior. Is this expected?

**Comment by junegunn** [comment]
I can't explain why, but adding `zle reset-prompt` will help in that case.

https://github.com/junegunn/fzf/blob/8b0e1f941a419258b0877d9a5f71373d76472043/shell/key-bindings.zsh#L65
